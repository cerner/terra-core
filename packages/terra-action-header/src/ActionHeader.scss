local {
  .right-buttons > * {
    margin-bottom: 0;
    margin-left: 1.429rem;
    margin-right: 0;
    margin-top: 0;
  }

  .left-buttons > * {
    margin-bottom: 0;
    margin-left: 0;
    margin-right: 1.429rem;
    margin-top: 0;
  }
}
