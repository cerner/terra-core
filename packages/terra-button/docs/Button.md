| Prop Name | Type | Is Required | Default Value | Description | 
|-|-|-|-|-|
| children | `node`  | optional |  | Child Nodes |
| href | `string`  | optional |  | Sets the href. When set will render the component as an anchor tag |
| icon | `element`  | optional |  | An optional icon. Nested inline with the text when provided |
| isBlock | `bool`  | optional | `false` | Whether or not the button should display as a block |
| isCompact | `bool`  | optional | `false` | Whether or not the button has reduced padding |
| isDisabled | `bool`  | optional | `false` | Whether or not the button should be disabled |
| isReversed | `bool`  | optional | `false` | Reverses the position of the icon and text |
| onClick | `func`  | optional |  | Callback function triggered when clicked |
| size | `enum('tiny','small','medium','large','huge')`  | optional |  | Sets the button size. One of tiny, small, medium, large, huge |
| text | `string`  | optional |  | Sets the button text |
| variant | `enum('default','link','primary','secondary')`  | optional | `'default'` | Sets the button variant. One of primary, secondary, or link |