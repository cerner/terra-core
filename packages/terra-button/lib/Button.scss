@import '~terra-mixins';
@import './variables';
@import './mixins';

// Button Component
// ====================================================

// html input[type='button'].terra-Button selector used to
// prevent a WebKit bug destroys native `audio` and `video` controls
// in Android 4. More info:
// https://github.com/necolas/normalize.css/issues/20
// https://code.google.com/p/android/issues/detail?id=34301
// appearance: button; set to correct the inability to style
// clickable types in iOS and Safari.
// Appearance needs to be applied only to button/input elements, as
// applying to anchor element will cause render issue.
// More info:
// https://github.com/necolas/normalize.css/issues/566
// https://github.com/necolas/normalize.css/commit/1be44f3b2a79af5d463c7ea0264fee0e6fa464ca
// stylelint-disable selector-no-qualifying-type
button.terra-Button,
html input[type='button'].terra-Button,
input[type='reset'].terra-Button,
input[type='submit'].terra-Button {
  appearance: button;  // autoprefix fix into -webkit-appearance
}
// stylelint-enable selector-no-qualifying-type

.terra-Button {
  @include terra-margin-horizontal($terra-button-margin-left, $terra-button-margin-right);
  @include terra-padding-horizontal($terra-button-padding-left, $terra-button-padding-right);
  @include terra-text-align-center;

  border-radius: 0.25em;
  border-style: solid;
  border-width: $terra-button-border-width;
  cursor: pointer;
  display: inline-block;
  font-family: $terra-font-family;
  font-size: map-get($terra-button-font-sizes, medium);
  font-weight: $terra-button-font-weight;
  line-height: 1.429; // 20px when html font size is computed to 14px
  margin-bottom: $terra-button-margin-bottom;
  margin-top: $terra-button-margin-top;
  padding-bottom: $terra-button-padding-bottom;
  padding-top: $terra-button-padding-top;
  text-decoration: none;
  text-transform: $terra-button-text-transform;
  touch-action: manipulation; // Enable fast tap interactiion in webkit browsers; see https://webkit.org/blog/5610/more-responsive-tapping-on-ios/
  user-select: none; // Prevent text selection on buttons on mobile devices
  vertical-align: middle;
  white-space: normal;

  // Removes extra space Firefox adds to button elements
  &::-moz-focus-inner {
    border: 0;
    padding: 0;
  }

  &:-moz-focusring {
    outline: 2px solid -moz-mac-focusring;
    -moz-outline-radius: 0.45em;
  }

  &:disabled,
  &.is-disabled {
    cursor: default;
    opacity: 0.7;
    pointer-events: none; // prevent pointer interaction for anchor elements
  }

  // Remove margin used to add space between buttons
  &:last-child {
    @include terra-margin-end(0);
  }
}

.terra-Button-text:first-child:not(:last-child) {
  @include terra-margin-end($terra-button-child-margin);
}

.terra-Button-text:nth-child(2) {
  @include terra-margin-start($terra-button-child-margin);
}

// Button modifiers
.terra-Button--default {
  @include terra-button-color($terra-button-default-color-scheme);
}

.terra-Button--primary {
  @include terra-button-color($terra-button-primary-color-scheme);
}

.terra-Button--secondary {
  @include terra-button-secondary($terra-button-secondary-color-scheme);
}

.terra-Button--link {
  @include terra-button-link($terra-button-link-color-scheme);
}

// Button size modifiers
.terra-Button--tiny {
  font-size: map-get($terra-button-font-sizes, tiny);
}

.terra-Button--small {
  font-size: map-get($terra-button-font-sizes, small);
}

.terra-Button--medium {
  font-size: map-get($terra-button-font-sizes, medium);
}

.terra-Button--large {
  font-size: map-get($terra-button-font-sizes, large);
}

.terra-Button--huge {
  font-size: map-get($terra-button-font-sizes, huge);
}

// Button style modifiers
.terra-Button--block {
  display: block;
  width: 100%;
}

.terra-Button--compact {
  padding-bottom: $terra-button-compact-padding-top;
  padding-top: $terra-button-compact-padding-bottom;
}
