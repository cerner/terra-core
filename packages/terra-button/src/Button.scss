@import './variables';
@import './mixins';

:local {
  /* stylelint-disable selector-class-pattern  */

  // Button Component
  // ====================================================

  // html input[type='button'].terra-Button selector used to
  // prevent a WebKit bug destroys native `audio` and `video` controls
  // in Android 4. More info:
  // https://github.com/necolas/normalize.css/issues/20
  // https://code.google.com/p/android/issues/detail?id=34301
  // appearance: button; set to correct the inability to style
  // clickable types in iOS and Safari.
  // Appearance needs to be applied only to button/input elements, as
  // applying to anchor element will cause render issue.
  // More info:
  // https://github.com/necolas/normalize.css/issues/566
  // https://github.com/necolas/normalize.css/commit/1be44f3b2a79af5d463c7ea0264fee0e6fa464ca
  // stylelint-disable selector-no-qualifying-type
  button.button,
  html input[type='button'].button,
  input[type='reset'].button,
  input[type='submit'].button {
    appearance: button;  // autoprefix fix into -webkit-appearance
  }
  // stylelint-enable selector-no-qualifying-type

  .button {
    border-radius: var(--terra-button-border-radius, 0.25em);
    border-style: solid;
    border-width: var(--terra-button-border-width, 1px);
    cursor: pointer;
    display: inline-block;
    font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
    font-size: map-get($terra-button-font-sizes, medium);
    font-weight: var(--terra-button-font-weight, 400);
    line-height: 1.429; // 20px when html font size is computed to 14px
    padding-bottom: var(--terra-button-padding-bottom, 0.286em);
    padding-left: var(--terra-button-padding-left, 0.714em);
    padding-right: var(--terra-button-padding-right, 0.714em);
    padding-top: var(--terra-button-padding-top, 0.286em);
    text-align: center;
    text-decoration: none;
    text-transform: var(--terra-button-text-transform, none);
    touch-action: manipulation; // Enable fast tap interactiion in webkit browsers; see https://webkit.org/blog/5610/more-responsive-tapping-on-ios/
    user-select: none; // Prevent text selection on buttons on mobile devices
    vertical-align: middle;
    white-space: normal;

    // Remove the inner border and padding in Firefox.
    &::-moz-focus-inner {
      border: 0;
      padding: 0;
    }

    // Restore the focus styles unset by the previous rule.
    &:-moz-focusring {
      outline: 1px dotted #1c1f21;
      -moz-outline-radius: 0.25em;
    }

    &:disabled,
    &.is-disabled {
      cursor: default;
      opacity: 0.7;
      pointer-events: none; // prevent pointer interaction for anchor elements
    }
  }

  // Button modifiers
  .default {
    @include terra-button-color($terra-button-default-color-scheme);
  }

  .primary {
    @include terra-button-color($terra-button-primary-color-scheme);
  }

  .secondary {
    @include terra-button-secondary($terra-button-secondary-color-scheme);
  }

  .link {
    @include terra-button-link($terra-button-link-color-scheme);
  }

  // Button size modifiers
  .tiny {
    font-size: map-get($terra-button-font-sizes, tiny);
  }

  .small {
    font-size: map-get($terra-button-font-sizes, small);
  }

  .medium {
    font-size: map-get($terra-button-font-sizes, medium);
  }

  .large {
    font-size: map-get($terra-button-font-sizes, large);
  }

  .huge {
    font-size: map-get($terra-button-font-sizes, huge);
  }

  // Button style modifiers
  .block {
    display: block;
    width: 100%;
  }

  .compact {
    padding-bottom: var(--terra-button-padding-bottom-compact, 0);
    padding-top: var(--terra-button-padding-top-compact, 0);
  }
}
