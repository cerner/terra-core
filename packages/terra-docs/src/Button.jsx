import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ThemeContext from 'terra-theme-context';

import styles from './Button.module.scss';

const cx = classNames.bind(styles);

/**
 * Enables focus styles for the target of the given event. Typically used as an onBlur callback on selectable elements.
 */
const enableFocusStyles = (event) => {
  event.currentTarget.setAttribute('data-focus-styles-enabled', 'true');
};

/**
 * Disables focus styles for the target of the given event. Typically used as an onMouseDown callback on selectable elements.
 */
const disableFocusStyles = (event) => {
  event.currentTarget.setAttribute('data-focus-styles-enabled', 'false');
};

const propTypes = {
  /**
  * The button text.
  */
  children: PropTypes.string,
};

const Button = ({
  children,
  ...customProps
}) => {
  const theme = React.useContext(ThemeContext);

  const buttonClassNames = classNames(
    cx([
      'button',
      theme.className,
    ]),
    customProps.className,
  );

  return (
    <button
      {...customProps}
      type="button"
      className={buttonClassNames}
      onBlur={enableFocusStyles}
      onMouseDown={disableFocusStyles}
      data-focus-styles-enabled
    >
      {children}
    </button>
  );
};

Button.propTypes = propTypes;

export default Button;
