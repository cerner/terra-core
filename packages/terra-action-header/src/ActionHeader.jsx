import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Button, { ButtonVariants } from 'terra-button';
import { FormattedMessage } from 'react-intl';
import ActionHeaderContainer from './_ActionHeaderContainer';
import styles from './ActionHeader.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Displays a single terra `Collapsible Menu View` (_Not provided by `Action Header`_) child element on the right end of the header.
   */
  children: PropTypes.element,
  /**
   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=1`. This helps screen readers to announce appropriate heading levels.
   * Changing 'level' will not visually change the style of the content.
   */
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  /**
   * Callback function for when the close button is clicked.
   * On small viewports, this will be triggered by a back button if onBack is not set.
   */
  onClose: PropTypes.func,
  /**
   * Callback function for when the back button is clicked. The back button will not display if this is not set.
   */
  onBack: PropTypes.func,
  /**
   * Callback function for when the expand button is clicked.
   * The expand button will not display if this is not set or on small viewports.
   * Only the expand button will be rendered if onMaximize and onMinimize are set.
   *
   * *Note: If `onBack` is set, the maximize button will not appear and a custom maximize button must be provided
   * as a child inside a `Collapsible Menu View`.*
   */
  onMaximize: PropTypes.func,
  /**
   * Callback function for when the minimize button is clicked.
   * The minimize button will not display if this is not set or on small viewports.
   * Only the expand button will be rendered if both onMaximize and onMinimize are set.
   *
   * *Note: If `onBack` is set, the minimize button will not appear and a custom minimize button must be provided
   * as a child inside a `Collapsible Menu View`.*
   */
  onMinimize: PropTypes.func,
  /**
   * Callback function for when the next button is clicked. The previous-next button group will display if either this or onPrevious is set but the button for the one not set will be disabled.
   */
  onNext: PropTypes.func,
  /**
   * Callback function for when the previous button is clicked. The previous-next button group will display if either this or onNext is set but the button for the one not set will be disabled.
   */
  onPrevious: PropTypes.func,
  /**
   * Text to be displayed as the title in the header bar.
   */
  title: PropTypes.string,
};

const defaultProps = {
  title: undefined,
  level: 1,
  onClose: undefined,
  onBack: undefined,
  onMaximize: undefined,
  onMinimize: undefined,
  onNext: undefined,
  onPrevious: undefined,
  children: undefined,
};

const ActionHeader = ({
  title,
  level,
  onClose,
  onBack,
  onMaximize,
  onMinimize,
  onPrevious,
  onNext,
  children,
  ...customProps
}) => {
  const closeButton = onClose
    ? (
      <FormattedMessage id="Terra.actionHeader.close">
        {closeText => (
          <Button
            className={cx(['header-button', 'close-button'])}
            data-terra-action-header="close-button"
            isIconOnly
            icon={<span className={cx(['header-icon', 'close'])} />}
            text={closeText}
            onClick={onClose}
            variant={ButtonVariants.UTILITY}
          />
        )}
      </FormattedMessage>
    )
    : null;
  const backButton = onBack
    ? (
      <FormattedMessage id="Terra.actionHeader.back">
        {backText => (
          <Button
            className={cx(['header-button', 'back-button'])}
            data-terra-action-header="back-button"
            isIconOnly
            icon={<span className={cx(['header-icon', 'back'])} />}
            text={backText}
            onClick={onBack}
            variant={ButtonVariants.UTILITY}
          />
        )}
      </FormattedMessage>
    )
    : null;

  let expandButton;
  if (!backButton) {
    if (onMaximize) {
      expandButton = (
        <FormattedMessage id="Terra.actionHeader.maximize">
          {maximizeText => (
            <Button
              className={cx(['header-button', 'maximize-button'])}
              data-terra-action-header="maximize-button"
              isIconOnly
              icon={<span className={cx(['header-icon', 'maximize'])} />}
              text={maximizeText}
              onClick={onMaximize}
              variant={ButtonVariants.UTILITY}
            />
          )}
        </FormattedMessage>
      );
    } else if (onMinimize) {
      expandButton = (
        <FormattedMessage id="Terra.actionHeader.minimize">
          {minimizeText => (
            <Button
              className={cx(['header-button', 'minimize-button'])}
              data-terra-action-header="minimize-button"
              isIconOnly
              icon={<span className={cx(['header-icon', 'minimize'])} />}
              text={minimizeText}
              onClick={onMinimize}
              variant={ButtonVariants.UTILITY}
            />
          )}
        </FormattedMessage>
      );
    }
  }

  const previousNextButtonGroup = (onPrevious || onNext)
    ? (
      <div className={cx('previous-next-button-group')}>
        <FormattedMessage id="Terra.actionHeader.previous">
          {previousText => (
            <Button
              className={cx(['header-button', 'previous-button'])}
              data-terra-action-header="previous-button"
              isIconOnly
              icon={<span className={cx(['header-icon', 'previous'])} />}
              text={previousText}
              onClick={onPrevious}
              isDisabled={onPrevious === undefined}
              variant={ButtonVariants.UTILITY}
            />
          )}
        </FormattedMessage>
        <FormattedMessage id="Terra.actionHeader.next">
          {nextText => (
            <Button
              className={cx(['header-button', 'next-button'])}
              data-terra-action-header="next-button"
              isIconOnly
              icon={<span className={cx(['header-icon', 'next'])} />}
              text={nextText}
              onClick={onNext}
              isDisabled={onNext === undefined}
              variant={ButtonVariants.UTILITY}
            />
          )}
        </FormattedMessage>

      </div>
    )
    : null;

  const leftButtons = (backButton || expandButton || previousNextButtonGroup)
    ? (
      <div className={cx('left-buttons')}>
        {backButton}
        {expandButton}
        {previousNextButtonGroup}
      </div>
    )
    : null;

  const rightButtons = closeButton ? <div className={cx('right-buttons')}>{closeButton}</div> : null;

  return (
    <ActionHeaderContainer
      {...customProps}
      startContent={leftButtons}
      title={title}
      endContent={rightButtons}
      level={level}
    >
      {children}
    </ActionHeaderContainer>
  );
};

ActionHeader.propTypes = propTypes;
ActionHeader.defaultProps = defaultProps;

export default ActionHeader;
