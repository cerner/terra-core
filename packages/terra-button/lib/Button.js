'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('./Button.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * Child Nodes
   */
  children: _react.PropTypes.node,
  /**
   * Sets the href. When set will render the component as an anchor tag
   */
  href: _react.PropTypes.string,
  /**
   * An optional icon. Nested inline with the text when provided
   */
  icon: _react.PropTypes.element,
  /**
   * Whether or not the button should display as a block
   */
  isBlock: _react.PropTypes.bool,
  /**
   * Whether or not the button has reduced padding
   */
  isCompact: _react.PropTypes.bool,
  /**
   * Whether or not the button should be disabled
   */
  isDisabled: _react.PropTypes.bool,
  /**
   * Reverses the position of the icon and text
   */
  isReversed: _react.PropTypes.bool,
  /**
   * Callback function triggered when clicked
   */
  onClick: _react.PropTypes.func,
  /**
   * Sets the button size. One of tiny, small, medium, large, huge
   */
  size: _react.PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
  /**
   * Sets the button text
   */
  text: _react.PropTypes.string,
  /**
   * Sets the button variant. One of primary, secondary, or link
   */
  variant: _react.PropTypes.oneOf(['default', 'link', 'primary', 'secondary'])
};

var defaultProps = {
  isBlock: false,
  isCompact: false,
  isDisabled: false,
  isReversed: false,
  variant: 'default'
};

var Button = function Button(_ref) {
  var children = _ref.children,
      icon = _ref.icon,
      isBlock = _ref.isBlock,
      isCompact = _ref.isCompact,
      isDisabled = _ref.isDisabled,
      isReversed = _ref.isReversed,
      size = _ref.size,
      text = _ref.text,
      variant = _ref.variant,
      customProps = _objectWithoutProperties(_ref, ['children', 'icon', 'isBlock', 'isCompact', 'isDisabled', 'isReversed', 'size', 'text', 'variant']);

  var attributes = _extends({}, customProps);
  var buttonText = text ? _react2.default.createElement(
    'span',
    { className: 'terra-Button-text' },
    text
  ) : null;

  attributes.className = (0, _classnames2.default)(['terra-Button', 'terra-Button--' + variant, { 'is-disabled': isDisabled }, _defineProperty({}, 'terra-Button--' + size, size), { 'terra-Button--block': isBlock }, { 'terra-Button--compact': isCompact }, attributes.className]);

  attributes.disabled = isDisabled;
  attributes.tabIndex = isDisabled ? '-1' : undefined;
  attributes['aria-disabled'] = isDisabled;

  var order = isReversed ? [buttonText, icon, children] : [icon, buttonText, children];

  return _react2.default.createElement.apply(_react2.default, [attributes.href ? 'a' : 'button', attributes].concat(order));
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

exports.default = Button;