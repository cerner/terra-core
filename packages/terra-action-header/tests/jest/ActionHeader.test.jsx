/* eslint-disable no-alert, no-console */
import React from 'react';
import Button from 'terra-button';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import ActionHeader from '../../src/ActionHeader';

describe('ActionHeader', () => {
  // Snapshot Tests
  it('should render a default action header', () => {
    const actionHeader = <ActionHeader />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with title', () => {
    const actionHeader = <ActionHeader title="Action Header" />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with back button and title', () => {
    const actionHeader = <ActionHeader title="Action Header" onBack={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with close button and title', () => {
    const actionHeader = <ActionHeader title="Action Header" onClose={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with back and close buttons and title', () => {
    const actionHeader = <ActionHeader title="Action Header" onBack={() => {}} onClose={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with level three header element and title', () => {
    const actionHeader = <ActionHeader title="Action Header" level={3} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with custom button and title', () => {
    const actionHeader = <ActionHeader title="Action Header"><Button text="Custom Button" onClick={() => alert('You clicked me!')} /></ActionHeader>;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with multiple custom buttons and title', () => {
    const actionHeader = (
      <ActionHeader title="Action Header">
        <span>
          <Button text="Custom Button" onClick={() => alert('You clicked me!')} />
          <Button text="Custom Button" onClick={() => alert('You clicked me!')} />
        </span>
      </ActionHeader>
    );
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with maximize button and title', () => {
    const actionHeader = <ActionHeader title="Action Header" onMaximize={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with minimize button and title', () => {
    const actionHeader = <ActionHeader title="Action Header" onMinimize={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with next and previous buttons and title', () => {
    const actionHeader = <ActionHeader title="Action Header" onNext={() => {}} onPrevious={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with title, enabled next button, and disabled previous button', () => {
    const actionHeader = <ActionHeader title="Action Header" onNext={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an action header with title, enabled previous button, and disabled next button', () => {
    const actionHeader = <ActionHeader title="Action Header" onPrevious={() => {}} />;
    const wrapper = shallowWithIntl(actionHeader);
    expect(wrapper).toMatchSnapshot();
  });
});
