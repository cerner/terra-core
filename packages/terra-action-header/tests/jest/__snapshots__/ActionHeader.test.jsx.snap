// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionHeader should render a default action header 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={null}
/>
`;

exports[`ActionHeader should render an action header with back and close buttons and title 1`] = `
<ActionHeaderContainer
  endContent={
    <div
      className="right-buttons"
    >
      <Button
        className="header-close-button"
        icon={
          <span
            className="header-icon close"
          />
        }
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={true}
        isReversed={false}
        onClick={[Function]}
        text="Close"
        type="button"
        variant="neutral"
      />
    </div>
  }
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <Button
        className="header-back-button"
        icon={
          <span
            className="header-icon back"
          />
        }
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={true}
        isReversed={false}
        onClick={[Function]}
        text="Back"
        type="button"
        variant="neutral"
      />
    </div>
  }
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with back button and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <Button
        className="header-back-button"
        icon={
          <span
            className="header-icon back"
          />
        }
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={true}
        isReversed={false}
        onClick={[Function]}
        text="Back"
        type="button"
        variant="neutral"
      />
    </div>
  }
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with close button and title 1`] = `
<ActionHeaderContainer
  endContent={
    <div
      className="right-buttons"
    >
      <Button
        className="header-close-button"
        icon={
          <span
            className="header-icon close"
          />
        }
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={true}
        isReversed={false}
        onClick={[Function]}
        text="Close"
        type="button"
        variant="neutral"
      />
    </div>
  }
  level={1}
  startContent={null}
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with custom button and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={null}
  title="Action Header"
>
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Custom Button"
    type="button"
    variant="neutral"
  />
</ActionHeaderContainer>
`;

exports[`ActionHeader should render an action header with level three header element and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={3}
  startContent={null}
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with maximize button and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <Button
        icon={
          <span
            className="header-icon maximize"
          />
        }
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={true}
        isReversed={false}
        onClick={[Function]}
        text="Maximize"
        type="button"
        variant="neutral"
      />
    </div>
  }
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with minimize button and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <Button
        icon={
          <span
            className="header-icon minimize"
          />
        }
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={true}
        isReversed={false}
        onClick={[Function]}
        text="Minimize"
        type="button"
        variant="neutral"
      />
    </div>
  }
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with multiple custom buttons and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={null}
  title="Action Header"
>
  <span>
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Custom Button"
      type="button"
      variant="neutral"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Custom Button"
      type="button"
      variant="neutral"
    />
  </span>
</ActionHeaderContainer>
`;

exports[`ActionHeader should render an action header with next and previous buttons and title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <ButtonGroup
        isBlock={false}
        selectedKeys={Array []}
      >
        <ButtonGroupButton
          icon={
            <span
              className="header-icon previous"
            />
          }
          isDisabled={false}
          onClick={[Function]}
          text="Previous"
        />
        <ButtonGroupButton
          icon={
            <span
              className="header-icon next"
            />
          }
          isDisabled={false}
          onClick={[Function]}
          text="Next"
        />
      </ButtonGroup>
    </div>
  }
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with title 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={null}
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with title, enabled next button, and disabled previous button 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <ButtonGroup
        isBlock={false}
        selectedKeys={Array []}
      >
        <ButtonGroupButton
          icon={
            <span
              className="header-icon previous"
            />
          }
          isDisabled={true}
          text="Previous"
        />
        <ButtonGroupButton
          icon={
            <span
              className="header-icon next"
            />
          }
          isDisabled={false}
          onClick={[Function]}
          text="Next"
        />
      </ButtonGroup>
    </div>
  }
  title="Action Header"
/>
`;

exports[`ActionHeader should render an action header with title, enabled previous button, and disabled next button 1`] = `
<ActionHeaderContainer
  endContent={null}
  level={1}
  startContent={
    <div
      className="left-buttons"
    >
      <ButtonGroup
        isBlock={false}
        selectedKeys={Array []}
      >
        <ButtonGroupButton
          icon={
            <span
              className="header-icon previous"
            />
          }
          isDisabled={false}
          onClick={[Function]}
          text="Previous"
        />
        <ButtonGroupButton
          icon={
            <span
              className="header-icon next"
            />
          }
          isDisabled={true}
          text="Next"
        />
      </ButtonGroup>
    </div>
  }
  title="Action Header"
/>
`;
